src/arrayMethodSort.js

```diff
@@ -5,7 +5,31 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    if (compareFunction) {
+      for (let i = 0; i < this.length; i++) {
+        for (let j = i + 1; j < this.length; j++) {
+          if (compareFunction(this[i], this[j]) > 0) {
+            const temp = this[i];
+
+            this[i] = this[j];
+            this[j] = temp;
+          }
+        }
+      }
+    } else {
+      for (let i = 0; i < this.length; i++) {
+        for (let j = i + 1; j < this.length; j++) {
+          if (this[i].toString() > this[j].toString()) {
```

Comment:
Export this condition as a separate function and use it as the default value of `compareFunction`, then there will be no need to add an if condition and duplicate similar code